docs/content/docs/dev/datastream/dataset_migration.md

```diff
@@ -0,0 +1,758 @@
+        });
+        </code>
+    </td>
+    <td>
+        <code style="white-space: pre-line;">DataStream&ltTuple2&ltString, Integer&gt&gt dataStream = // [...]
+dataStream.keyBy(value -> value.f0)
+        .reduce(new ReduceFunction(){
+            // implement user-defined reduce logic
+        });
+        </code>
+    </td>
+  </tr>
+</table>
+
+### Aggregate on Grouped DataSet
+
+<table>
+  <tr>
+    <th>DataSet</th>
+    <th>DataStream</th>
+  </tr>
+  <tr>
+    <td>
+        <code style="white-space: pre-line;">DataSet&ltTuple2&ltString, Integer&gt&gt dataSet = // [...]
+// compute sum of the second field
+dataSet.groupBy(value -> value.f0).aggregate(SUM, 1);
+// compute min of the second field
+dataSet.groupBy(value -> value.f0).aggregate(MIN, 1);
+// compute max of the second field
+dataSet.groupBy(value -> value.f0).aggregate(MAX, 1);
+        </code>
+    </td>
+    <td>
+        <code style="white-space: pre-line;">DataStream&ltTuple2&ltString, Integer&gt&gt dataStream = // [...]
+// compute sum of the second field
+dataStream.keyBy(value -> value.f0).sum(1);
+// compute min of the second field
+dataStream.keyBy(value -> value.f0).min(1);
+// compute max of the second field
+dataStream.keyBy(value -> value.f0).max(1);
+        </code>
+    </td>
+  </tr>
+</table>
+
+#### Category 2
+
+For category 2, these DataSet APIs can be implemented by DataStream APIs with different semantic but same calculation behavior.
+The developers need to adapt their code to accommodate these variations, which introduces additional complexity.
+
+### Distinct
+
+<table>
+  <tr>
+    <th>DataSet</th>
+    <th>DataStream</th>
+  </tr>
+  <tr>
+    <td>
+        <code style="white-space: pre-line;">DataSet&ltInteger&gt dataSet = // [...]
+dataSet.distinct();
+        </code>
+    </td>
+    <td>
+        <code style="white-space: pre-line;">DataStream&ltInteger&gt dataStream = // [...]
+dataStream
+        .keyBy(value -> value)
+        .reduce((value1, value2) -> value1);
+        </code>
+    </td>
+  </tr>
+</table>
+
+### Hash-Partition
+
+<table>
+  <tr>
+    <th>DataSet</th>
+    <th>DataStream</th>
+  </tr>
+  <tr>
+    <td>
+        <code style="white-space: pre-line;">DataSet&ltTuple2&ltString, Integer&gt&gt dataSet = // [...]
+dataSet.partitionByHash(value -> value.f0);
+        </code>
+    </td>
+    <td>
+        <code style="white-space: pre-line;">DataStream&ltTuple2&ltString, Integer&gt&gt dataStream = // [...]
+// partition by the hashcode of key
+dataStream.partitionCustom((key, numSubpartition) -> key.hashCode() % numSubpartition, value -> value.f0);
+        </code>
+    </td>
+  </tr>
+</table>
+
+### Reduce on Full DataSet
+
+If developers want to compute data of full datastream, `GlobalWindow` could be used to collect all records of datastream.
+However, a special trigger is also required to trigger the computation of `GlobalWindow` at the end of its inputs. Here is an example 
+code snippet of the trigger.
```

Comment:
How about rewriting this paragraph as follows?

> Operations on a full DataSet correspond to the global window aggregation in DataStream with a custom trigger that invokes the computation of `GlobalWindow` at the end of the inputs. Below is an example code snippet that will be reduced in the rest of this document.

Reply:
Done.